Source/Core/VideoCommon/GeometryShaderManager.cpp

```diff
@@ -45,21 +47,45 @@ void GeometryShaderManager::SetConstants()
   {
     s_projection_changed = false;
 
+    constants.stereoparams.fill(0);
+
     if (xfmem.projection.type == GX_PERSPECTIVE)
     {
-      float offset = (g_ActiveConfig.iStereoDepth / 1000.0f) *
-                     (g_ActiveConfig.iStereoDepthPercentage / 100.0f);
-      constants.stereoparams[0] = g_ActiveConfig.bStereoSwapEyes ? offset : -offset;
-      constants.stereoparams[1] = g_ActiveConfig.bStereoSwapEyes ? -offset : offset;
+      if (auto* const openxr_session = g_renderer->GetOpenXRSession())
+      {
+        Common::Matrix44 projection;
+        std::memcpy(projection.data.data(), VertexShaderManager::constants.projection.data(),
+                    sizeof(projection));
+
+        // We must "undo" the game's projection before applying our eye projections.
+        const auto inv_projection = projection.Inverted();
+
+        // Use the game's near and far values.
+        const float z_near = projection.data[11] / (projection.data[10] - 1);
+        const float z_far = projection.data[11] / (projection.data[10] + 1);
+
+        int eye_index = 0;
+        for (auto& eye_view : constants.eye_views)
+        {
+          eye_view = openxr_session->GetEyeViewMatrix(eye_index, z_near, z_far) * inv_projection;
+          ++eye_index;
+        }
+      }
+      else
+      {
+        float offset = (g_ActiveConfig.iStereoDepth / 1000.0f) *
+                       (g_ActiveConfig.iStereoDepthPercentage / 100.0f);
+        constants.stereoparams[0] = g_ActiveConfig.bStereoSwapEyes ? offset : -offset;
+        constants.stereoparams[1] = -constants.stereoparams[0];
+        constants.stereoparams[2] = (float)(g_ActiveConfig.iStereoConvergence *
```

Comment:
`static_cast` or `float(expression)`

Reply:
I moved existing code but I can fix it. :3